### ISSUE TITLE
Make `switch` option more useful

### ISSUE THREAD

**Issue by aghreed** [gstroup/apimocker#43]
I suppose this is more of a general question about the usefulness of the `switch` option for a specific service configuration.

Let's take this example:

```
    "users/:userId": {
      "mockFile": "users/users.json",
      "verbs": ["get", "put"],
      "switch": "userId",
      "responses": {
        "put": {"httpStatus": 200, "mockFile": "users/updated.user.josn"}
      }
    }
```

Wouldn't it be great if what `switch` did was to look inside the specified `mockFile` of `users/users.json`, iterate the array to find where the value of the `userId` property matched the `userId` parameter sent in the request, then returned that single user object out of the array of users?

From what I can tell looking at apimocker's source, this is totally possible. If you think its a decent idea, I'd be more than willing to make a PR.

Thoughts?

**Comment by gstroup** [comment]
The original design was to put one response in each mock file.  Take a look in the /samplemocks directory.  So the switch will look for a matching file, i.e. a matching user.  Is there a reason you want to put all your responses (users) in a single mock file?

**Comment by end-user** [comment]
I'm also looking for that functionality.  The design of one file per query doesn't scale well if you have a variety of situations to test and limits the utility of the module.  It looks like it's really only designed to deal with a handful of cases that might represent unit testing.  This limits its ability to handle a broad array of requests when the criteria get diverse.  What if you wanted to load a table of 20 keys, and each add an additional couple of options?  You're now managing a directory with nearly a hundred files, each with their own values that you need to be correct (for testing).  It should be a trivial matter to load one file and perform a sort or search to match the lookup value.  However, this does start to look a lot like a database at that point.

**Comment by gstroup** [comment]
Can you please take a look at PR #50 ?  I believe this addresses your request.  I'm hoping to merge this in soon.  Thanks.

**Comment by gstroup** [comment]
Closing this issue, since it has been implemented in PR #50 .
